src/rum/core.cljs

```diff
@@ -39,12 +39,12 @@
 
         ctor (fn [props]
                (this-as this
-                 (set! (.-state this)
-                       #js {":rum/state"
-                            (-> (gobj/get props ":rum/initial-state")
-                                (assoc :rum/react-component this)
-                                (call-all init props)
-                                volatile!)})
+                 (gobj/set this "state"
```

Comment:
I'd say it's "future proof". Also `aset` is going to be deprecated at some point and property name symbols like `-state` in `set!` are mungled by GCC, unless externs provided
```